### PR TITLE
disabling audience validation in the JWT Validation struct

### PR DESCRIPTION
**Pull request by moulimukherjee**
### Motivation
Fixing failing nightlies
More context: https://materializeinc.slack.com/archives/CTESPM7FU/p1703815384845419

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - 

cc @def- @MaterializeInc/frontend

### CODE REVIEW

File: src/frontegg-client/src/error.rs [hunk 1]
```diff
@@ -75,6 +75,6 @@ pub enum Error {
     /// An error indicating that the claims are invalid, acoording to the structure
     /// or keys provided. If the error persists,
     /// check if the token claims matches the struct.
-    #[error("Decoding JWT claims.")]
```
Comment (moulimukherjee) [inline src/frontegg-client/src/error.rs:78]: cc @joacoc I kept this from your PR

File: ci/nightly/pipeline.template.yml [hunk 2]
```diff
@@ -827,7 +827,7 @@ steps:
     plugins:
       - ./ci/plugins/mzcompose:
           composition: cloud-canary
-    branches: "main v*.*"
```
Comment (moulimukherjee) [inline ci/nightly/pipeline.template.yml:830]: This is temporary to just kick off a build, will remove this commit when it's time to commit.

Comment (moulimukherjee) [reply ci/nightly/pipeline.template.yml:830]: https://buildkite.com/materialize/nightlies/builds/5762

File: src/frontegg-client/src/error.rs (hunk 1 above)
Comment (joacoc) [reply src/frontegg-client/src/error.rs:78]: @moulimukherjee thank you!

**Review by joacoc — APPROVED** [review]

File: ci/nightly/pipeline.template.yml (hunk 2 above)
Comment (def-) [reply ci/nightly/pipeline.template.yml:830]: As the nightly was green I'll be so free to remove this: https://buildkite.com/materialize/nightlies/builds/5764

**Review by jkosh44 — APPROVED** [review]

**Pull request merged**